src/app/campaign/flight/flight.container.ts

```diff
@@ -13,50 +13,68 @@ import { map } from 'rxjs/operators';
       [flight]="flightLocal$ | async"
       (flightUpdate)="flightUpdateFromForm($event)"
     ></grove-flight>
+    <grove-availability
+      [flight]="flightLocal$ | async"
+      [availabilityZones]="flightAvailability$ | async">
+    </grove-availability>
   `,
+  styleUrls: ['flight.container.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class FlightContainerComponent implements OnInit {
+export class FlightContainerComponent implements OnInit, OnDestroy {
   private currentFlightId: string;
-  state$ = new ReplaySubject<FlightState>(1);
-  flightLocal$ = this.state$.pipe(map(state => state.localFlight));
+  flightState$ = new ReplaySubject<FlightState>(1);
+  flightLocal$ = this.flightState$.pipe(map((state: FlightState) => state.localFlight));
+  flightAvailability$ = this.campaignStoreService.currentFlightAvailability$;
   currentInventoryUri$ = new ReplaySubject<string>(1);
   inventoryOptions$: Observable<Inventory[]>;
   zoneOptions$: Observable<InventoryZone[]>;
+  flightSub: Subscription;
 
   constructor(
     private route: ActivatedRoute,
     private inventoryService: InventoryService,
     private campaignStoreService: CampaignStoreService,
     private router: Router
-  ) {
-    this.route.paramMap.subscribe(params => this.setFlightId(params.get('flightid')));
+  ) {}
+
+  ngOnInit() {
+    this.flightSub = combineLatest(this.route.paramMap, this.campaignStoreService.campaign$).subscribe(([params, campaignState]) => {
```

Comment:
I needed to change this from a) being a nested subscription and b) using `first()` of campaign$. Because `first()` completes after the first update, it broke the unidirectional flow of data. campaign$ is basically root state for this container, and that state is passed down to child components. Child components like the new availability table were not getting updates to the store state because the subscription was cut off by first().